### PR TITLE
Starting point for #20 Validate all POST data

### PR DESCRIPTION
**Pull request by zero-24**
@mbabker can this be a starting point for #20 or is this to unspecific?

### CODE REVIEW

File: src/Controllers/SubmitControllerCreate.php [hunk 1]
```diff
@@ -88,4 +117,47 @@ function ($value)
...
+		$dotchecks = explode('.' $data);
+
+		// We ever use 2 dots and 3 parts in our CMS Version
+		if (count($dotchecks) != 3)
+		{
+			return false;
+		}
+
+		// The pulugin is installed since 3.5.0 other CMS Versions can't have this plugin installed
+		if (version_compare($data, '3.5.0', '<'))
```
Comment (wilsonge) [inline src/Controllers/SubmitControllerCreate.php]: We may as well also put less than Joomla 4 as well

Comment (zero-24) [inline src/Controllers/SubmitControllerCreate.php]: see: https://github.com/joomla-extensions/jstats-server/commit/b3be6627a22dfcc3598a1fcfe56274c7d3f76abf

Comment (wilsonge) [inline src/Controllers/SubmitControllerCreate.php]: :+1:

**Comment by wilsonge** [comment]
We can do sanity checks on PHP Version as well I know our min PHP version is 5.3.10 - but to keep things simple I'd say php version above 5.3.0 and less than 8.0.0

**Comment by zero-24** [comment]
Is this a Unit test that fails? https://travis-ci.org/joomla-extensions/jstats-server/jobs/91593845 Or do i something wrong here?

File: src/Controllers/SubmitControllerCreate.php [hunk 2]
```diff
@@ -88,4 +117,53 @@ function ($value)
...
+	/**
+	 * Check the Database type
+	 *
+	 * @return  false on failiure else the Database type
+	 *
+	 * @since   1.0
+	 */
+	private function checkDatabaseType($data)
+	{
+		if (!in_array($data, $databaseTypes))
```
Comment (wilsonge) [inline src/Controllers/SubmitControllerCreate.php]: `$this->databaseTypes` otherwise you get the unit test failure

**Comment by wilsonge** [comment]
yup unit test failure - see my comment

File: src/Controllers/SubmitControllerCreate.php (hunk 1 above)
Comment (mbabker) [inline src/Controllers/SubmitControllerCreate.php]: I wouldn't block 3.x at all.  Fake or not, there's a 3.2.1 install in the database.  Doing this blocks being able to distro the plugin as an optional thing for earlier releases.  Doubt anyone would add it, but it keeps the door open.

**Comment by zero-24** [comment]
>We can do sanity checks on PHP Version as well I know our min PHP version is 5.3.10 - but to keep things simple I'd say php version above 5.3.0 and less than 8.0.0

Done with: https://github.com/joomla-extensions/jstats-server/commit/4c8ed92d1bac121a493613ba7d9858b6aab28cad

File: src/Controllers/SubmitControllerCreate.php (hunk 1 above)
Comment (wilsonge) [inline src/Controllers/SubmitControllerCreate.php]: If we're doing that then we need to allow 5.3.3 and higher for PHP Version (cause of the minimum php version before 3.3)

Comment (zero-24) [inline src/Controllers/SubmitControllerCreate.php]: We are now open for 3.0.0 - 4.0.0: https://github.com/joomla-extensions/jstats-server/commit/9208d2f0a5403990c1c4c94c56ed42763b8395b8

Comment (mbabker) [inline src/Controllers/SubmitControllerCreate.php]: PHP 5.3.1 was the 3.0.0 minimum.

File: src/Controllers/SubmitControllerCreate.php [hunk 3]
```diff
@@ -88,4 +118,78 @@ function ($value)
...
+	 * Check the PHP Version
+	 *
+	 * @return  false on failiure else the PHP Version
+	 *
+	 * @since   1.0
+	 */
+	private function checkPHPVersion($data)
+	{
+		// We only support < 5.3.10 ...
+		if (version_compare($data, '5.3.10', '<'))
```
Comment (wilsonge) [inline src/Controllers/SubmitControllerCreate.php]: 5.3.1 then here

Comment (zero-24) [inline src/Controllers/SubmitControllerCreate.php]: done: https://github.com/joomla-extensions/jstats-server/commit/8d56b80244e97243c1a6edd450704cdef84d0529

File: src/Controllers/SubmitControllerCreate.php (hunk 1 above)
Comment (zero-24) [inline src/Controllers/SubmitControllerCreate.php]: Yes i have updated that as well here: https://github.com/joomla-extensions/jstats-server/commit/8d56b80244e97243c1a6edd450704cdef84d0529 Thanks.

**Comment by mbabker** [comment]
I'm pulling this local and squashing it with a little cleanup.  Give me a bit of time.

**Comment by phproberto** [comment]
If we are going to add validations we should return null here when there are no matches:

https://github.com/joomla-extensions/jstats-server/blob/c4b8e10ed1f6ba9c2935f2126d040f64336fcb11/src/Controllers/SubmitControllerCreate.php#L74

File: src/Controllers/SubmitControllerCreate.php [hunk 4]
```diff
@@ -88,4 +119,78 @@ function ($value)
...
+	/**
+	 * Check the CMS Version
+	 *
+	 * @return  false on failiure else the CMS Version
+	 *
+	 * @since   1.0
+	 */
+	private function checkCMSVersion($data)
+	{
+		$dotchecks = explode('.', $data);
```
Comment (phproberto) [inline src/Controllers/SubmitControllerCreate.php]: See my comment on the PR. We don't need this if the preg_match function does that job for us

**Comment by mbabker** [comment]
Agreed, adding that to my cleanup.

**Comment by mbabker** [comment]
Squashed and committed.  Just need some unit test cases now for the validations.

**Pull request closed without merging**